assets/js/googlesitekit/data/utils.js

```diff
@@ -377,3 +379,24 @@ export function createValidationSelector( validate ) {
 		dangerousSelector,
 	};
 }
+
+/**
+ * Creates a validated action creator.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Function} validate      A function for validating action arguments.
+ * @param {Function} actionCreator A function for returning or yielding redux-style actions.
+ * @return {Function} An enhanced action creator.
+ */
+export function createValidatedAction( validate, actionCreator ) {
+	invariant( typeof validate === 'function', 'a validator function is required.' );
+	invariant( typeof actionCreator === 'function', 'an action creator function is required.' );
+	invariant( ! isGenerator( validate ), 'an action’s validator function must not be a generator.' );
```

Comment:
@eugene-manuilov Good point on it being a private API! I didn't spot that. I had to install this package in order for Jest to use it so I can revert that change

This was already on @aaemnnosttv's PR (and the IB was only to add to the PR :upside_down_face: )

I'm making your requested change anyway @eugene-manuilov  but just tagged Evan so he can see :eyes: 

My only concern was that maybe in the future there's a better way to check for generator functions but I doubt it will change :smile: